### PR TITLE
Alerting: Add filters for RouteGetRuleStatuses

### PR DESCRIPTION
**Pull request by fayzal-g**
**What is this feature?**
Adds filters for `folder_uid`, `rule_group` and `rule_name` in the `/api/prometheus/grafana/api/v1/rules` endpoint, for example: `/api/prometheus/grafana/api/v1/rules?folder_uid=folder&rule_group=group&rule_name=name`

**Why do we need this feature?**
Allows the optimization of the frontend to be able to render the View and Edit rule pages for GMA much faster (see https://github.com/grafana/grafana/pull/87625)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by stevesg — COMMENTED** [review]
LGTM but I think the splitting by comma will break if the rule name or group has a comma in it. Luckily url.Values will decode multiple arguments for us.

File: pkg/services/ngalert/api/api_prometheus.go [hunk 1]
```diff
@@ -292,16 +295,26 @@ func PrepareRuleGroupStatuses(log log.Logger, manager state.AlertInstanceManager
...
+	_, exists := opts.Namespaces[folderUID]
+	if folderUID != "" && exists {
+		namespaceUIDs = append(namespaceUIDs, folderUID)
+	} else {
+		for k := range opts.Namespaces {
+			namespaceUIDs = append(namespaceUIDs, k)
+		}
 	}
 
+	ruleGroups := opts.Query["rule_group"]
```
Comment (JohnnyQQQQ) [inline pkg/services/ngalert/api/api_prometheus.go:310]:
```suggestion
```

File: pkg/services/ngalert/api/api_prometheus.go [hunk 2]
```diff
@@ -292,16 +295,26 @@ func PrepareRuleGroupStatuses(log log.Logger, manager state.AlertInstanceManager
...
+
 	alertRuleQuery := ngmodels.ListAlertRulesQuery{
 		OrgID:         opts.OrgID,
 		NamespaceUIDs: namespaceUIDs,
 		DashboardUID:  dashboardUID,
 		PanelID:       panelID,
+		RuleGroups:    ruleGroups,
```
Comment (JohnnyQQQQ) [inline pkg/services/ngalert/api/api_prometheus.go:317]:
```suggestion
		RuleGroups:    opts.Query["rule_group"],
```

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM, I think you can directly pass the query parameters and don't need to write them first into another variable.

File: pkg/services/ngalert/api/api_prometheus.go [hunk 3]
```diff
@@ -235,6 +235,9 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *contextmodel.ReqContext) respon
 	return response.JSON(ruleResponse.HTTPStatusCode(), ruleResponse)
 }
 
+// TODO: Refactor this function to reduce the cylomatic complexity
```
Comment (stevesg) [inline pkg/services/ngalert/api/api_prometheus.go:238]: Raise an issue for this so we don't forget :)

**Review by stevesg — APPROVED** [review]
Looks good

**Pull request merged**